### PR TITLE
Remove SQL callout cookie from JIT.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
See https://github.com/dotnet/coreclr/pull/26458#issuecomment-527286662 for issue.
See https://github.com/dotnet/coreclr/pull/26458#issuecomment-539145589 for separate PR.

Will post `jit-diff` results presently.

/cc @jkotas @AndyAyersMS

### CODE REVIEW

**Comment by jkotas** [comment]
Could you please include the matching ABI doc update with this as well?

File: src/jit/morph.cpp [hunk 1]
```diff
@@ -2669,19 +2669,24 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
...
-        assert(!call->gtCallCookie);
-        // Add a conservative estimate of the stack size in a special parameter (r11) at the call site.
-        // It will be used only on the intercepted-for-host code path to copy the arguments.
+    CLANG_FORMAT_COMMENT_ANCHOR;
 
-        GenTree* cns     = new (this, GT_CNS_INT) GenTreeIntCon(TYP_I_IMPL, fgEstimateCallStackSize(call));
-        call->gtCallArgs = gtPrependNewCallArg(cns, call->gtCallArgs);
-        numArgs++;
+#if !defined(FEATURE_CORECLR)
+        if (call->IsUnmanaged() && !opts.ShouldUsePInvokeHelpers())
```
Comment (jkotas) [inline src/jit/morph.cpp]: Can this be indented one to the left? Or does the formatter insist on it being like this?

Comment (AaronRobinsonMSFT) [reply src/jit/morph.cpp]: Formatter insists on that.

**Comment by AaronRobinsonMSFT** [comment]
```
jit-diff diff --pmi --base [BASE]\Product\Windows_NT.x64.Checked --diff [DIFF]\Product\Windows_NT.x64.Checked --core_root [BASE]\tests\Windows_NT.x64.Release\Tests\Core_Root --frameworks --tsv
Using --output D:\coreclr\bin\diffs
Beginning PMI Diffs for System.Private.CoreLib.dll, framework assemblies
| Finished 129/129 Base 129/129 Diff [395.9 sec]
Completed PMI Diffs for System.Private.CoreLib.dll, framework assemblies in 395.94s
Diffs (if any) can be viewed by comparing: [BASE]\diffs\dasmset_2\base [BASE]\diffs\dasmset_2\diff
Analyzing diffs...
Found 33 files with textual diffs.
PMI Diffs for System.Private.CoreLib.dll, framework assemblies for  default jit
Summary:
(Lower is better)
Total bytes of diff: -2651 (-0.01% of base)
    diff is an improvement.
Top file improvements by size (bytes):
       -1704 : System.Private.CoreLib.dasm (-0.04% of base)
        -257 : System.Console.dasm (-0.45% of base)
        -225 : System.IO.FileSystem.dasm (-0.20% of base)
         -77 : Microsoft.Diagnostics.Tracing.TraceEvent.dasm (-0.00% of base)
         -75 : System.Diagnostics.Process.dasm (-0.08% of base)
33 total files with size differences (33 improved, 0 regressed), 96 unchanged.
Top method regressions by size (bytes):
          12 ( 1.89% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,int,int) (3 methods)
           5 ( 2.25% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,bool,bool,struct)
           4 ( 0.88% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,ref,int):int (2 methods)
           4 ( 1.71% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,ref,int)
           4 ( 1.71% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,ref,int):int
Top method improvements by size (bytes):
         -31 (-1.77% of base) : System.Console.dasm - ConsolePal:SetWindowSize(int,int)
         -29 (-1.15% of base) : System.Console.dasm - ConsolePal:MoveBufferArea(int,int,int,int,int,int,ushort,int,int)
         -24 (-0.33% of base) : System.Diagnostics.Process.dasm - Process:StartWithCreateProcess(ref):bool:this (2 methods)
         -24 (-1.90% of base) : System.Private.CoreLib.dasm - WindowsVersion:GetIsWindows8OrAbove():bool (2 methods)
         -21 (-0.77% of base) : System.IO.FileSystem.dasm - FileSystemEnumerator`1:FindNextEntry():this (7 methods)
Top method regressions by size (percentage):
           5 ( 2.25% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,bool,bool,struct)
          12 ( 1.89% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,int,int) (3 methods)
           4 ( 1.71% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,ref,int)
           4 ( 1.71% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,ref,int):int
           3 ( 1.44% of base) : System.Private.CoreLib.dasm - ILStubClass:IL_STUB_PInvoke(struct,int,int,long)
Top method improvements by size (percentage):
          -9 (-3.64% of base) : System.Console.dasm - ConsolePal:set_BackgroundColor(int)
          -7 (-3.50% of base) : System.Collections.Concurrent.dasm - PlatformHelper:get_ProcessorCount():int
          -7 (-3.50% of base) : System.Threading.Tasks.Parallel.dasm - PlatformHelper:get_ProcessorCount():int
         -12 (-3.17% of base) : System.Private.CoreLib.dasm - EventPipeEventProvider:System.Diagnostics.Tracing.IEventProvider.EventWriteTransfer(long,byref,long,long,long,int,long):int:this
          -9 (-3.15% of base) : System.Private.CoreLib.dasm - CompareInfo:FindStringOrdinal(int,ref,int,int,ref,int,bool):int
557 total methods with size differences (545 improved, 12 regressed), 202387 unchanged.
Completed analysis in 33.80s
```

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib

**Pull request merged**